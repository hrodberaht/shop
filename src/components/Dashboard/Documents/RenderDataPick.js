import React from 'react';
import DatePicker from 'react-datepicker';
import ShowErrorInForm from './ShowErrorInForm';

const RenderDataPick = ({ input: { onChange, value }, meta }) => (
  <React.Fragment>
    <DatePicker
      className="form-input"
      selected={value}
      onChange={onChange}
      dateFormat="yyyy/MM/dd"
    />
    <ShowErrorInForm className="error-text" {...meta} />
  </React.Fragment>
);

export default RenderDataPick;
