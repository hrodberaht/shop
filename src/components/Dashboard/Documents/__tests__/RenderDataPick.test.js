import RenderDataPick from '../RenderDataPick';

const defaultProps = {
  input: {
    onChange: jest.fn()
  }
};

const setup = buildSetup(RenderDataPick, defaultProps);

describe('<RenderDataPick />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
