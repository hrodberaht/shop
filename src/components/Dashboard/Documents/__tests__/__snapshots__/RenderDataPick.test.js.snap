// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RenderDataPick /> should render without crash 1`] = `
<Fragment>
  <DatePicker
    allowSameDay={false}
    className="form-input"
    dateFormat="yyyy/MM/dd"
    dateFormatCalendar="LLLL yyyy"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    monthsShown={1}
    nextMonthButtonLabel="Next month"
    onBlur={[Function]}
    onChange={[MockFunction]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onInputClick={[Function]}
    onInputError={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    preventOpenOnFocus={false}
    previousMonthButtonLabel="Previous Month"
    readOnly={false}
    renderDayContents={[Function]}
    shouldCloseOnSelect={true}
    showTimeSelect={false}
    timeCaption="Time"
    timeIntervals={30}
    withPortal={false}
  />
  <ShowErrorInForm
    className="error-text"
    error={null}
    touched={false}
  />
</Fragment>
`;
